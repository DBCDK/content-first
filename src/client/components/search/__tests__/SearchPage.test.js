import React from 'react';
import {
  SearchPage,
  mapStateToProps,
  mapDispatchToProps
} from '../SearchPage.container';
import renderer from 'react-test-renderer';

describe('SearchPage', () => {
  describe('render function', () => {
    it('displays search results', () => {
      const tree = renderer
        .create(
          <SearchPage
            searching={false}
            query={'søgeforespørgsel'}
            results={[{pid: '123:samplepid-456', title: 'Titel', creator: 'Forfatter'}]}
            onSearch={() => {}}
          />
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('gives a message, when no search results are present', () => {
      const tree = renderer
        .create(
          <SearchPage
            searching={false}
            query={'søgeforespørgsel'}
            onSearch={() => {}}
          />
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  it('has a working mapStateToProps', () => {
    expect(
      mapStateToProps({
        searchReducer: {
          query: 'q',
          loading: 'l',
          results: []
        }
      })
    ).toEqual({
      query: 'q',
      searching: 'l',
      results: []
    });
  });
  it('has a working mapDispatchToProps', () => {
    expect(Object.keys(mapDispatchToProps(() => {}))).toEqual(['onSearch']);
  });
});
