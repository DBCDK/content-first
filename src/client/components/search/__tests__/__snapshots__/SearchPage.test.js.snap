// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPage render function displays search results 1`] = `
<div
  className="col-xs-11 col-centered text-left"
>
  <form
    className="input-group"
    onSubmit={[Function]}
    style={
      Object {
        "marginBottom": 30,
        "marginTop": 40,
      }
    }
  >
    <input
      className="form-control"
      placeholder="Søg titel/forfatter"
    />
    <span
      className="input-group-btn"
    >
      <button
        className="btn btn-default"
        onClick={[Function]}
        type="button"
      >
        🔍
      </button>
    </span>
  </form>
  <div>
    <p>
      Søgning efter "
      søgeforespørgsel
      " returnerede 
      1
       
      resultater.
    </p>
    <ul>
      <li>
        Titel
         - 
        Forfatter
      </li>
    </ul>
  </div>
</div>
`;

exports[`SearchPage render function gives a message, when no search results are present 1`] = `
<div
  className="col-xs-11 col-centered text-left"
>
  <form
    className="input-group"
    onSubmit={[Function]}
    style={
      Object {
        "marginBottom": 30,
        "marginTop": 40,
      }
    }
  >
    <input
      className="form-control"
      placeholder="Søg titel/forfatter"
    />
    <span
      className="input-group-btn"
    >
      <button
        className="btn btn-default"
        onClick={[Function]}
        type="button"
      >
        🔍
      </button>
    </span>
  </form>
  <div>
    <p>
      Søgning efter "
      søgeforespørgsel
      " returnerede 
      1
       
      resultater.
    </p>
    <ul>
      <li>
        Titel
         - 
        Forfatter
      </li>
    </ul>
  </div>
</div>
`;
